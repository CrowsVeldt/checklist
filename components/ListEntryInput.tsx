import { StyleSheet, TextInput, View } from "react-native";

export default function ListEntryInput() {
  return (
    <View style={styles.inputWrapper}>
      <TextInput defaultValue="Entry Title" />
    </View>
  );
}

const styles = StyleSheet.create({
  inputWrapper: {
    flexDirection: "row",
    alignItems: "center",
  },
});
