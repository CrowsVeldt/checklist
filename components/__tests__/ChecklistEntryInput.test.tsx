import { render } from "@testing-library/react-native";
import ListEntryInput from "../ChecklistEntryInput";

describe("<ListEntryInput />", () => {
  test("ListEntryInput renders correctly", () => {
    const tree = render(
      <ListEntryInput
        id="abc"
        initialTitle="title"
        onEntryChange={() => console.log("change")}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
