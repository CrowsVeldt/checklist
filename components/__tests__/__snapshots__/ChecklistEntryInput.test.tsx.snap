// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListEntryInput /> ListEntryInput renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "width": "70%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    value="title"
  />
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text>
      Required?
    </Text>
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "#657786",
            "borderRadius": 2,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          {
            "borderRadius": 5,
          },
          false,
          false,
          undefined,
          false,
        ]
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={{}}
  >
    <Text />
  </View>
</View>
`;
